src/main/java/gift/dto/kakaomessage/Content.java

```diff
@@ -0,0 +1,17 @@
+package gift.dto.kakaomessage;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+
+@JsonNaming(SnakeCaseStrategy.class)
+public record Content(
+        String title,
+        String imageUrl,
+        String description,
+        Link link
+) {
+    public Content(String title, String imageUrl, String description, String commonUrl) {
+        this(title, imageUrl, description, new Link(commonUrl));
+    }
```

Comment:
```suggestion
    Content content = new Content(title, imageUrl, description, new Link(commonUrl));
```
String을 받아서 생성자에서 Link를 생성하는 것이랑 Link를 생성해서 주입하는 것이랑 어떤 차이점이 있을까요? 또한 각각의 장단점이 무엇일까요? :thinking:

Reply:
Content라는 dto에서 Link라는 dto의 구조 또는 생성자 사용법까지 알아야한다는 단점이 있을것 같습니다.
만약 Link의 구조가 변경될 경우, Link부터 Service레이어까지 모든 dto의 구조를 변경해야할 수도 있겠습니다.
특히, Link는 실제로 WebUrl과 MobileUrl 2개가 존재하는데, 해당 생성자부에서 하나의 인자로 2개의 멤버변수에 할당하고있는 상태입니다. 이 2개를 개별값으로 지정하고자 하는 경우가 위에서 언급한 경우가 되겠습니다.

장점으로는 현재와 같이, 단순하게 생성자를 사용할 수 있다는 점이 있겠습니다.

결론적으로, 책임분리 및 확장성을 고려해서 **호출부에서 Link 생성 후 주입**하는 방식으로 변경하는것이 좋아보입니다!